src/util.cpp

```diff
@@ -890,6 +890,20 @@ bool NetworkUtil::ParseNetworkAddress ( QString       strAddress,
 }
 
 
+CHostAddress NetworkUtil::GetLocalAddress()
+{
+    QTcpSocket socket;
+    socket.connectToHost("8.8.8.8", 53); // google DNS, or something else reliable
+    if (socket.waitForConnected()) {
+        return CHostAddress( socket.localAddress(), 0 );
+    } else {
+        qWarning()
```

Comment:
Where does this goes? If you are on Linux, do you see it on the command line? Is it needed also for the Windows user to get this warning? But the problem is that we have an auto-start of the Jamulus server on Windows. If we would get a warning message dialog, this might interfere with the auto start. So I guess it is better to silently set it to localhost. I assume in 99.9999 % of the cases we can successfully get the local address.

Reply:
If you're starting up the server, it's logged along with the usual start up messages.

And yes, it's highly unlikely to fail and the fallback to localhost is at least safe.

The only other thing I could think of was trying the given central server on port 80 - but if you're running your own central server, it might not have port 80 open.  So there's not really much improvement.

I hope that socket disconnects when it goes out of scope - I forgot to check that.